lib/agent.js

```diff
@@ -26,7 +26,9 @@ const Test = require('./test.js');
 function TestAgent(app, options = {}) {
   if (!(this instanceof TestAgent)) return new TestAgent(app, options);
 
-  Agent.call(this, options);
+  const agent = new Agent(options);
```

Comment:
Looks like this commit to change from functions to classes breaks this behavior.
https://github.com/ladjs/superagent/commit/fca95a3e0b0de12a96e6fffbc486efcb267883e6

The `apply` hook in the Proxy seems to discard the `this` reference and so it not being binded.

Reply:
This looks promising (and explanation makes sense) 👀